recipes/zlib-ng/all/conanfile.py

```diff
@@ -0,0 +1,89 @@
+from conans import ConanFile, CMake
+from conans.tools import download, unzip, replace_in_file, remove_files_by_mask, get
+import os
+
+class ZlibNgConan(ConanFile):
+    name = "zlib-ng"
+    description = "zlib data compression library for the next generation systems"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Dead2/zlib-ng/"
+    license ="Zlib"
+    topics = ("conan", "zlib", "compression")
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        get(**self.conan_data["sources"][self.version])
+        os.rename("zlib-ng-{}".format(self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+        self._cmake.definitions["ZLIB_COMPAT"] = False
+        self._cmake.definitions["ZLIB_ENABLE_TESTS"] = False
+        self._cmake.definitions["WITH_GZFILEOP"] = True
+        self._cmake.definitions["WITH_OPTIM"] = True
+        self._cmake.definitions["WITH_NEW_STRATEGIES"] = True
+        self._cmake.definitions["WITH_NATIVE_INSTRUCTIONS"] = False
+        self._cmake.definitions["WITH_SANITIZER"] = False
+        self._cmake.definitions["WITH_FUZZERS"] = False
+        self._cmake.definitions["WITH_MAINTAINER_WARNINGS"] = False
+        self._cmake.definitions["WITH_CODE_COVERAGE"] = False
+
+        self._cmake.configure(build_folder=self._build_subfolder, source_folder=self._source_subfolder)
```

Comment:
By setting the source folder you are likely missing the Conan settings... take a look at our `cmake_wrapper` https://github.com/conan-io/conan-center-index/blob/master/recipes/base64/all/CMakeLists.txt 


```suggestion
        self._cmake.configure(build_folder=self._build_subfolder)
```